### PR TITLE
name: last(split(resourceId, '/'))! is fixed by removing exclamation mark

### PR DESCRIPTION
**Pull request by paggarwalmsft**
# Description

>Thank you for your contribution !

name: last(split(resourceId, '/'))! here exclamation mark was causing error "Error BCP236: Expected a new line or comma character at this location."

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/aad/domain-services/.bicep/nested_roleAssignments.bicep [hunk 1]
```diff
@@ -53,7 +53,7 @@ var builtInRoleNames = {
 }
 
 resource AzureADDS 'Microsoft.AAD/DomainServices@2021-05-01' existing = {
-  name: last(split(resourceId, '/'))!
+  name: last(split(resourceId, '/'))
```
Comment (eriqua) [inline modules/aad/domain-services/.bicep/nested_roleAssignments.bicep:56]: Hey @paggarwalmsft, could you kindly provide more info on the error you are experiencing? E.g., is it when deploying locally or through pipelines? Which bicep version are you using to deploy?

Comment (AlexanderSehr) [reply modules/aad/domain-services/.bicep/nested_roleAssignments.bicep:56]: Echoing @eriquas comment, this character was added to align with a specific null-checker linter rule (see this [PR](https://github.com/Azure/ResourceModules/pull/2806)). That being said, maybe they changed their rules yet again.
Like Erika said, please let us know which Bicep CLI version you're using (`bicep -version`).

Comment (paggarwalmsft) [reply modules/aad/domain-services/.bicep/nested_roleAssignments.bicep:56]: Bicep CLI version 0.18.4 (1620479ac6)

**Comment by AlexanderSehr** [comment]
Following a joint investigation, we established that the issue was introduced with a new Linter Rule back in January ([ref PR](https://github.com/Azure/ResourceModules/issues/2641)). As the modules were updated in accordance with the new rule, older Bicep versions would not be compatible (which essentially leads to the raised warning).
After agreeing with @paggarwalmsft  I'll go ahead and close this PR as a result.

**Pull request closed without merging**